omod-common/src/main/java/org/openmrs/module/webservices/rest/util/UniqueIdentifierUtil.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.util;
+
+import java.util.regex.Pattern;
+
+/**
+ * The Universally Unique Identifier utility class. This class checks for a UUID's validity using a
+ * regular expression. because Java UUID class doesn't provide any function to validate whether a
+ * string is a valid UUID or not. This is particularly useful because of the need to Refactor
+ * TaskAction Resource to take in a Task Definition uuid in addition to name. Requests for resources
+ * must be made with valid UUIDs
+ */
+public class UniqueIdentifierUtil {
```

Comment:
I thought we would assume that the passed value is a uuid and if we fail to get a task definition by it, we revert to the old behaviour. Then would we need this class?

Reply:
Well, I needed to split the logic in for the method getResourceByNameOrUuid(String str) and this class helps validate a UUID. Only falling back to default implementation if what is parsed is not a UUID.

Reply:
What value does that add compared to just doing the default if the uuid returns null?